### PR TITLE
Use systems facilities api and fix clinic api call

### PR DESCRIPTION
**Pull request by jbalboni**
## Description
This integrates the DS facilities api and also fixes an issue with the clinic api, where we were trying to derive the system id from the facility id (which is not always possible).

Most of the changes in here are due to the backend flattening the institution object.

## Testing done
Local and unit testing

## Acceptance criteria
- [x] DS facilities call works

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by narin — APPROVED** [review]
Looks good.  Is the flattening being done by `vets-api` or did we discover a difference further down the chain?

**Comment by jbalboni** [comment]
@narin vets-api is doing the flattening, I believe.

**Comment by jbalboni** [comment]
Test coverage is low because of the mock data code that will be removed eventually.

File: src/applications/vaos/tests/actions/newAppointment.unit.spec.js [hunk 1]
```diff
@@ -191,19 +197,49 @@ describe('VAOS newAppointment actions', () => {
...
+            '323_983': facilities983Parsed,
+          },
+        },
+      };
+
+      const getState = () => previousState;
+
+      const thunk = openFacilityPage('vaFacility', {}, defaultSchema);
+      await thunk(dispatch, getState);
+      expect(dispatch.firstCall.args[0].type).to.equal(
```
Comment (U-DON) [inline src/applications/vaos/tests/actions/newAppointment.unit.spec.js]: Could this use the `succeededAction` defined below?

**Review by U-DON — APPROVED** [review]

**Pull request merged**